schema/resolvers.js

```diff
@@ -119,7 +119,42 @@ const Resolvers = {
     deleteOption: (_, args, ctx) =>
       ctx.repositories.Option.remove(args.input.id),
     undeleteOption: (_, args, ctx) =>
-      ctx.repositories.Option.undelete(args.input.id)
+      ctx.repositories.Option.undelete(args.input.id),
+    createOtherAnswer: async (root, args, ctx) => {
+      const parentAnswer = await ctx.repositories.Answer.get(
+        args.input.parentAnswerId
+      );
+
+      if (!isNil(parentAnswer.otherAnswerId)) {
+        return parentAnswer;
+      }
+
+      if (parentAnswer.type === "Radio" || parentAnswer.type === "Checkbox") {
+        const otherAnswer = await ctx.repositories.Answer.insert({
+          mandatory: false,
+          type: "TextField"
+        });
+        await ctx.repositories.Answer.update(
+          merge(parentAnswer, { otherAnswerId: otherAnswer.id })
+        );
+      }
+
+      return parentAnswer;
```

Comment:
Why return the parent and not the new answer?

Reply:
I was originally going to return the child, but decided against it. You can always get the child answer from the parent, but you can't get the parent from the child.

i.e.